File: codevs/codevs1400.pas
program cx;
uses math;
const dx:array[1..4]of longint=(1,-1,0,0);
      dy:array[1..4]of longint=(0,0,-1,1);
var i,j,n,m,ans,t,xx,yy:longint;
    g:array[0..1000,0..1000]of longint;

function gym(x,y:longint):longint;
var i:longint;
begin
  gym:=0;
  for i:=1 to 4 do
    if g[x,y]*g[x+dx[i],y+dy[i]]<0 then
      inc(gym,abs(g[x+dx[i],y+dy[i]]))
      else dec(gym,abs(g[x+dx[i],y+dy[i]]));
end;

begin
  read(n,m);
  while (m<>0)and(n<>0) do
  begin
    fillchar(g,sizeof(g),0);
    for i:=1 to n do
      for j:=1 to m do read(g[i,j]);
    ans:=-maxlongint;
    for i:=1 to n do
      for j:=1 to n do
        begin
          t:=gym(i,j);
          if t>ans then
            begin
              ans:=t;
              xx:=i; yy:=j;
            end;
        end;
    writeln(xx,' ',yy,' ',ans);
    read(n,m);
  end;
end.

File: codevs/codevs1400.cpp
#include<iostream>
#include<cstdio>
#include<algorithm>
using namespace std;
const int dx[]={0,1,-1,0,0};
const int dy[]={0,0,0,-1,1};
int i,j,n,m,ans,xx,yy,g[1000][1000];

int fuck(int x,int y){
  int haha=0;
  for (int k=1;k<=4;k++)
    if (g[x][y]*g[x+dx[k]][y+dy[k]]<0)
      haha+=abs(g[x+dx[k]][y+dy[k]]);
      else haha-=abs(g[x+dx[k]][y+dy[k]]);
  return haha;
}

int main(){
  while(cin>>n>>m&&n&&m){
    memset(g,0,sizeof(g));

    for (int i=1;i<=n;i++)
      for (int j=1;j<=m;j++) cin>>g[i][j];
    ans=-999999;
    for (i=1;i<=n;i++)
      for (j=1;j<=m;j++){
        int t=fuck(i,j);
        if (t>ans){ans=t;xx=i;yy=j;}
    }
    cout<<xx<<" "<<yy<<" "<<ans<<endl;
  }
  return 0;
}
